File: cpp/CyclicRotation.cpp
vector<int> solution(vector<int> &A, int K) {
    int s = A.size();
    vector<int> result;
    
    if (s==0)
        return {};
    else
    {
        result.reserve(s);
        int idx = s-(K%s);

        for(int i = idx; i<s; i++)
            result.emplace_back(A[i]);
        for(int i = 0; i<idx; i++)
            result.emplace_back(A[i]);

        return result;
    }
}

File: python/CyclicRotation.py
def solution(A, K):
    result = []
    size = len(A)

    if size==0:
        return result
    else:
        idx = size - (K%size)

        for i in range(idx, size):
            result.append(A[i])

        for i in range(0, idx):
            result.append(A[i])

        return result
